probe-rs/src/architecture/arm/core/exception_handling/armv6m_armv7m_shared.rs

```diff
@@ -45,15 +65,130 @@ bitfield! {
     pub apsr_z_bit, _: 30;
     pub apsr_c_bit, _: 29;
     pub apsr_v_bit, _: 28;
+    /// Indicates if the stack was realigned to 8 byte boundary on exception entry.
+    pub stack_was_realigned, _:9;
     pub exception_number, _: 8,0;
 }
 
 /// Decode the exception information.
 pub(crate) fn exception_details(
-    exception_interface: &dyn ExceptionInterface,
+    exception_interface: &impl ExceptionInterface,
     memory_interface: &mut dyn MemoryInterface,
     stackframe_registers: &DebugRegisters,
+    debug_info: &DebugInfo,
 ) -> Result<Option<ExceptionInfo>, Error> {
+    let frame_return_address = get_stack_frame_return_address(stackframe_registers)?;
+
+    if ExcReturn(frame_return_address).is_exception_flag() == 0xF {
+        // This is an exception frame.
+        let raw_exception = exception_interface.raw_exception(stackframe_registers)?;
+        let registers = exception_interface.calling_frame_registers(
+            memory_interface,
+            stackframe_registers,
+            raw_exception,
+        )?;
+        let description =
+            exception_interface.exception_description(raw_exception, memory_interface)?;
+
+        let exception_frame_pc =
+            registers.get_register_value_by_role(&RegisterRole::ProgramCounter)?;
+
+        let mut handler_frame = StackFrame {
+            id: get_object_reference(),
+            function_name: description.clone(),
+            source_location: None,
+            registers,
+            pc: RegisterValue::U32(exception_frame_pc as u32),
+            frame_base: None,
+            is_inlined: false,
+            local_variables: None,
+            canonical_frame_address: None,
+        };
+
+        // A fault that is escalated to the priority of a HardFault retains the program counter value of the original fault,
+        // so we have to unwind the frame pointer that matches.
+        if raw_exception == 3 {
```

Comment:
Can we make this non raw and use the typed interface?

Reply:
The enum is different for armv6 and 7, and not useful in the shared module. At least I can't see how to do it :)